src/utility/eql_root.jl

```diff
@@ -0,0 +1,47 @@
+"""
+x = eql_root(a,b,c)
+    Numerically stable method for computing the positive root
+    of the quadratic polynomial -ax^2 - 2bx + c, a >= 0.
+
+in
+- a : The negative of the x^2 term. Must be positive.
+- b : Half the negative of the x term.
+- c : The constant term.
+
+out
+- x : The positive root which satisfies -ax^2 - 2bx + c.
+"""
+function eql_root(a,b,c)
+    #TODO: Ask fess if the equations will necessarily be solvable with reals.
```

Comment:
the way this code is used the determinant will always be nonnegative.
ok to throw an error if det. is negative.
so make the output real, of this eltype:
` promote_type(eltype(a),eltype(b),eltype(c),Float32)`

Reply:
I'm pretty sure that sqrt will make the output real as is, if given real inputs.

Reply:
So the sqrt function will just throw a negative input error as the code lies, and otherwise, it'll output a real.